src/git.zsh

```diff
@@ -361,6 +512,40 @@ _fzf_complete_git_parse_completing_option() {
     esac
 }
 
+_fzf_complete_remove_quotes_while_holding_emtpy() {
+    local tmp=(\\0${(Q)^${(z)@}})
+    echo ${(q+)tmp#\\0}
+}
+
+_fzf_complete_git_parse_argument() {
+    local arguments=(${(z)1})
+    local index=$2
+    local options_argument_required=(${(z)3})
+    shift 3
+
+    if (( ${#arguments} <= 2 )); then
+        return 1
+    fi
+
+    local i
+    local command_arguments=()
+    for i in {3..${#arguments}}; do
+        if [[ ${(Q)arguments[$i]} = -(#c1,2)* ]]; then
+            continue
+        fi
+
+        local previous_argument=$(_fzf_complete_git_parse_completing_option '' ${(Q)arguments[(( i - 1 ))]} $options_argument_required '' )
+        if [[ -n $previous_argument ]] && [[ ${options_argument_required[(r)$previous_argument]} = $previous_argument ]]; then
+            continue
+        fi
+
+        command_arguments+=${arguments[$i]}
+    done
+
+    echo ${command_arguments[$index]}
+    return $(( $index > ${#command_arguments} ))
```

Comment:
As this is arithmetically evaluated, you actually don't have to use `$` to indicate variables. Wierd though!

```suggestion
    return $(( index > #command_arguments ))
```